server/controllers/user.js

```diff
@@ -1,12 +1,12 @@
-const showUser = (req, res, next) => {
+const showUser = async (req, res, next) => {
   // check for logged in user
   const userId = req.user ? req.user.id : null;
   if (!userId)
     return res
       .status(401)
       .send("You are not authorized to access this resource");
   try {
-    const user = User.findById(userId).select("-password").exec();
+    const user = await User.findById(userId).exec().then(user => user.toObject());
```

Comment:
@skassam21 My understanding is the Query.exec returns a promise which resolves to a document which can have the toObject method called on it. 

Is this the correct data flow for using toObject method?

Reply:
I believe it should work actually if you just do this:
`const user = await User.findById(userId)`

I would try it and see if it works - I think user.toObject is implicitly called whenever the data is fetched from the database and turned into an object in Javascript

Reply:
I don't think you even need to call exec if you are using await syntax

Reply:
I think the `findById` is sugar for `findOne` which returns a Query object if a callback is not supplied, I'll leave this as is until I can test it. 
[source](https://github.com/Automattic/mongoose/blob/bc5f6ded5970212ba9f49096d630eb0a1c7a2124/lib/query.js#L2215)

Reply:
This looks like it works, I was actually looking at using toJSON instead, this could also work but I like how you have done it here:
```
 const user = await User.findById(userId);
return res.status(200).send(user); // requires setting the toJSON function instead of toObject
```